Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSBeamCentreFinder.py

```diff
@@ -126,11 +130,11 @@ def PyInit(self):
         self.declareProperty('Centre2', 0.0, direction=Direction.Output,
                              doc="The centre position found in the second dimension")
 
-    def PyExec(self):
+    def PyExec(self): # noqa: C901
```

Comment:
Is there any possibility of factoring out some of this function so that the complexity warning turn-off is not needed? I think the complexity goes of branching paths so shifting all the plotting if else try excepts into a different function might be enough?

Reply:
I was going to refactor this separately, but if moving only the plotting to a separate function works then I'll do that here